nixos/modules/services/networking/cloudflared.nix

```diff
@@ -0,0 +1,95 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.cloudflared;
+in
+{
+  meta.maintainers = with maintainers; [ bbigras ];
+
+  options.services.cloudflared = {
+    enable = mkEnableOption "Cloudflared Argo Tunnel client daemon";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.cloudflared;
+      defaultText = "pkgs.cloudflared";
+      description = "The package to use for Cloudflared";
+    };
+
+    tunnels = mkOption {
+      description = ''
+        Cloudflare tunnels
+      '';
+      type = types.attrsOf (types.submodule ({ name, ... }: {
+        options = {
+          tunnel = mkOption {
+            type = with types; nullOr str;
+            default = null;
+            description = ''
+              Tunnel ID
```

Comment:
Several of your `description`s are leaving me wanting more. Can you provide more context and explanation for the sysadmin?

Reply:
"Tunnel ID" is half-correct, you can specify a name too. Suggest "Tunnel ID or name."
https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/config